### PR TITLE
feat(CommunityPermissions): prevent adding tokens and ens names that are already chosen

### PR DESCRIPTION
**Pull request by micieslak**
### What does the PR do

Implements logic preventing from duplicating tokens and ens names in `Who holds` section when creating/editing community permissions.

Closes: #8817 

### Affected areas
Components related to `Community Permissions` functionality.

### Screenshot of functionality (including design for comparison)
- [x] I've checked the design and this PR matches it

[Kazam_screencast_00107.webm](https://user-images.githubusercontent.com/20650004/214429305-538ab21b-cbb9-4b80-85cd-8c92188cd7e5.webm)

### CODE REVIEW

**Review by caybro — APPROVED** [review]
Looks good

**Review by alexandraB99 — APPROVED** [review]
lgtm

**Review by alexjba — APPROVED** [review]
LGTM! 👍

**Review by noeliaSD — CHANGES_REQUESTED** [review]
Some comments here:

In `collectibles` menu:
- Should the check icon be in the first deep page or in the second one?  For me, the navigation icon should be always visible and then when you move to the deep level, the check is there.

Screenshot 2023-01-25 at 12 33 02
Screenshot 2023-01-25 at 12 33 12

- Check is missing in `thumbnails` view:
Design:

Screenshot 2023-01-25 at 12 39 38

App:

https://user-images.githubusercontent.com/97019400/214554532-ab9b7e46-c32c-4a73-bb4b-35167bafda6f.mov

File: ui/app/AppLayouts/Chat/controls/community/HoldingTypes.qml [hunk 1]
```diff
@@ -6,6 +6,6 @@ QtObject {
     }
 
     enum Mode {
-        Add, Update
+        Add, Update, UpdateOrRemove
```
Comment (noeliaSD) [inline ui/app/AppLayouts/Chat/controls/community/HoldingTypes.qml:9]: Is it something different `Update` or `UpdateOrRemove`? afaik they always go together in the designs
Could we keep just only 2 states/modes?

File: ui/app/AppLayouts/Chat/views/communities/CommunityNewPermissionView.qml [hunk 2]
```diff
@@ -87,6 +87,48 @@ StatusScrollView {
             }
             property bool isPrivateDirty: false
 
+            function getIndexOfKey(key) {
```
Comment (noeliaSD) [inline ui/app/AppLayouts/Chat/views/communities/CommunityNewPermissionView.qml:90]: Does it make sense to have these functions in `CommunityPermissionHelpers` instead?

**Comment by micieslak** [comment]
Thank you @noeliaSD for your findings. Everything fixed!

**Review by noeliaSD — COMMENTED** [review]
In general, LGTM!

Added some other minor comments in code.

File: ui/app/AppLayouts/Chat/controls/community/HoldingTypes.qml (hunk 1 above)
Comment (micieslak) [reply ui/app/AppLayouts/Chat/controls/community/HoldingTypes.qml:9]: Yes, when we reach given token via "+" that is already added, then it's Update (no remove button). When we reach added token via clicking on an item on the list, then it's "Update or remove" (+ remove button).

File: ui/app/AppLayouts/Chat/views/communities/CommunityNewPermissionView.qml (hunk 2 above)
Comment (micieslak) [reply ui/app/AppLayouts/Chat/views/communities/CommunityNewPermissionView.qml:90]: I think it's quite tightly coupled with handling of dirty values so I decided to keep it here now. We can reconsider it when refactoring dirty values to be handled in a more systematic way.

File: ui/app/AppLayouts/Chat/controls/community/HoldingTypes.qml (hunk 1 above)
Comment (noeliaSD) [reply ui/app/AppLayouts/Chat/controls/community/HoldingTypes.qml:9]: Ok!! I didn't get this difference! Thanks!!

**Review by noeliaSD — APPROVED** [review]

**Comment by noeliaSD** [comment]
> Thank you @noeliaSD for your findings. Everything fixed!

Thanks for the fixes!! Looks great!

Ready for QA testing!! :)

**Review by elina2015 — APPROVED** [review]
Tested and approved

**Pull request merged**